packages/fxa-admin-server/src/subscriptions/stripe.service.ts

```diff
@@ -0,0 +1,153 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Firestore } from '@google-cloud/firestore';
+import { Inject, Injectable, Provider } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { MozLoggerService } from 'fxa-shared/nestjs/logger/logger.service';
+import { PaymentConfigManager } from 'fxa-shared/payments/configuration/manager';
+import {
+  determineIapIdentifiers,
+  StripeHelper,
+  STRIPE_PRICE_METADATA,
+} from 'fxa-shared/payments/stripe';
+import { StripeFirestore } from 'fxa-shared/payments/stripe-firestore';
+import {
+  AbbrevPlan,
+  MozillaSubscriptionTypes,
+  SubscriptionType,
+} from 'fxa-shared/subscriptions/types';
+import {
+  capabilitiesClientIdPattern,
+  subscriptionProductMetadataBaseValidator,
+} from 'fxa-shared/subscriptions/validation';
+import { StatsD } from 'hot-shots';
+import Stripe from 'stripe';
+import { AppConfig } from '../config';
+
+export const StripeFactory: Provider<Stripe> = {
+  provide: 'STRIPE',
+  useFactory: (configService: ConfigService) => {
+    const stripeConfig = configService.get('subscriptions');
+    return new Stripe(stripeConfig.stripeApiKey, {
+      apiVersion: '2020-08-27',
+      maxNetworkRetries: 3,
+    });
+  },
+  inject: [ConfigService],
+};
+
+@Injectable()
+export class StripeService extends StripeHelper {
+  public override readonly stripe: Stripe;
+
+  protected override readonly stripeFirestore: StripeFirestore;
+
+  protected override readonly paymentConfigManager?: PaymentConfigManager;
+
+  protected readonly firestore: Firestore;
+
+  constructor(
+    logger: MozLoggerService,
+    configService: ConfigService<AppConfig>,
+    @Inject('METRICS') metrics: StatsD,
+    @Inject('STRIPE') stripe: Stripe,
+    @Inject('FIRESTORE') firestore: Firestore
+  ) {
+    const config = {
+      subscriptions: configService.get('subscriptions'),
+      authFirestore: configService.get('authFirestore'),
+      subhub: configService.get('subhub'),
+      redis: configService.get('redis'),
+    };
+
+    super(config, metrics, logger);
+
+    // Intentionally leaving this out for now.
```

Comment:
Ah, I am glad you brought this up! I did this earlier on development work and actually forgot that I wanted to call this out in the review. There were two reasons I did this initially.

1. For stripe purchase manager wasn’t needed. I only realized after your commetnd and working on the app store / play store integration that it is. I’ll add this back in asap, and try to figure out a way to get this guarded by a test case.
I was a bit worried that the listeners in the purchase manager wouldn’t function properly in the admin server since it’s behind a VPN. I wanted to check with someone about this.


BTW, I think this also speaks to the issues I had with testing the app and play store integrations. I feel like creating that test infrastructure was beyond the scope of the task at hand. If there is something that I can just piggyback off of, that'd be great.